### PR TITLE
added WithOverrideEmptySlice config flag

### PR DESCRIPTION
**Pull request by rusdevops**
I want to overwrite the null slice with an empty slice

### CODE REVIEW

**Comment by rusdevops** [comment]
It doesn't work. Need to help

File: merge.go [hunk 1]
```diff
@@ -26,11 +26,12 @@ func hasExportedField(dst reflect.Value) (exported bool) {
...
-	AppendSlice             bool
-	TypeCheck               bool
-	Transformers            Transformers
-	overwriteWithEmptyValue bool
+	Overwrite                    bool
+	AppendSlice                  bool
+	TypeCheck                    bool
+	Transformers                 Transformers
+	overwriteWithEmptyValue      bool
+	OverwriteSliceWithEmptyValue bool
```
Comment (vdemeester) [inline merge.go]: This needs to be un-exported (as we are using `WithOverrideEmptySlice`), same as `overwriteWithEmptyValue` is not exported

**Comment by rusdevops** [comment]
> @rusdevops thanks for the PR. Can you add a test case for it ?

added

**Comment by rusdevops** [comment]
Please create a release after the merge 👍

**Comment by rusdevops** [comment]
@vdemeester, can you review PR ? please 😄

**Comment by rusdevops** [comment]
@imdario please see PR if you can

**Comment by rv-jbaker** [comment]
This should resolve an issue we are seeing too. We have some state that looks like
```
Dest
{
  "test": {
    "a": []string{"testString"}
  }
}

Src
{
  "test": {
    "a": []string{}
  }
}
```
where we want an empty array of strings in the src to overwrite the destination path and remove string `testString` from the path `test.a`.

I couldn't find a way to set `overwriteWithEmptyValue` to true, because if I could that would also resolve the issue without needing this change.

@imdario or @rusdevops any insight on a way I could accomplish this or is this PR needed to make what we need happen?

**Pull request merged**